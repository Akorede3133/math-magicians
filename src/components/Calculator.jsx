import { useState } from 'react';
import Button from './Button';
import data from '../data';
import Screen from './Screen';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [calcObj, setCalcObj] = useState({
    total: null,
    next: null,
    operation: null,
  });
  const handleClick = (buttonName) => {
    const newObj = calculate(calcObj, buttonName);
    setCalcObj(newObj);
  };

  return (
    <div className=" w-[90%] md:w-[500px] mx-auto">
      <Screen value={calcObj.next || (calcObj.total || '0')} />
      <div className="grid grid-cols-4">
        {
        data.map((val) => (
          <Button
            key={val.name}
            handleClick={() => handleClick(val.name)}
            type={val.type}
            name={val.name}
          />
        ))
      }
      </div>
    </div>
  );
};

export default Calculator;
