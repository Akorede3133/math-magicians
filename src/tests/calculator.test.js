import { render } from '@testing-library/react';
import Calculator from '../pages/Calculator';

describe('Calculator', () => {
  test('renders correctly', () => {
    const { container } = render(<Calculator />);
    expect(container).toMatchSnapshot();
  });
});
