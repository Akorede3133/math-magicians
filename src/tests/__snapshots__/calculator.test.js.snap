// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator renders correctly 1`] = `
<div>
  <div
    class=" w-[90%] md:w-[80%] md:mt-10 mx-auto py-5 grid md:grid-cols-[1fr_2fr] gap-4 md:gap-10"
  >
    <div>
      <p
        class=" font-bold text-3xl"
      >
        Lets do some math!
      </p>
    </div>
    <div
      class=""
    >
      <div
        class=" w-full bg-slate-600 h-16 flex justify-end items-center text-white p-4 text-2xl"
        data-testid="screen"
      >
        0
      </div>
      <div
        class="grid grid-cols-4"
      >
        <button
          class="false bg-slate-300 py-4 border-l border-b"
          type="button"
        >
          AC
        </button>
        <button
          class="false bg-slate-300 py-4 border-l border-b"
          type="button"
        >
          +/-
        </button>
        <button
          class="false bg-slate-300 py-4 border-l border-b"
          type="button"
        >
          %
        </button>
        <button
          class="false  bg-orange-500 py-4 border-l border-b"
          type="button"
        >
          รท
        </button>
        <button
          class="false bg-slate-300 py-4 border-l border-b"
          type="button"
        >
          7
        </button>
        <button
          class="false bg-slate-300 py-4 border-l border-b"
          type="button"
        >
          8
        </button>
        <button
          class="false bg-slate-300 py-4 border-l border-b"
          type="button"
        >
          9
        </button>
        <button
          class="false  bg-orange-500 py-4 border-l border-b"
          type="button"
        >
          x
        </button>
        <button
          class="false bg-slate-300 py-4 border-l border-b"
          type="button"
        >
          4
        </button>
        <button
          class="false bg-slate-300 py-4 border-l border-b"
          type="button"
        >
          5
        </button>
        <button
          class="false bg-slate-300 py-4 border-l border-b"
          type="button"
        >
          6
        </button>
        <button
          class="false  bg-orange-500 py-4 border-l border-b"
          type="button"
        >
          -
        </button>
        <button
          class="false bg-slate-300 py-4 border-l border-b"
          type="button"
        >
          1
        </button>
        <button
          class="false bg-slate-300 py-4 border-l border-b"
          type="button"
        >
          2
        </button>
        <button
          class="false bg-slate-300 py-4 border-l border-b"
          type="button"
        >
          3
        </button>
        <button
          class="false  bg-orange-500 py-4 border-l border-b"
          type="button"
        >
          +
        </button>
        <button
          class=" col-span-2 bg-slate-300 py-4 border-l border-b"
          type="button"
        >
          0
        </button>
        <button
          class="false bg-slate-300 py-4 border-l border-b"
          type="button"
        >
          .
        </button>
        <button
          class="false  bg-orange-500 py-4 border-l border-b"
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
